src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -297,7 +297,17 @@ class CosmasGoogleCloudService(private val freeBucketName: String,
             responseObserver.onError(StatusException(errorStatus))
             return@logging
         }
-        response.addAllVersions(versionList)
+        val actualVersionList = mutableListOf<FileVersionInfo>()
+        for (version in versionList) {
+            // checking that files from window really exist - storage.get doesn't load all file content
+            val blob = this.storage.get(getBlobId(version.fileId, request.info, version.generation))
```

Comment:
I thought we discussed fetching only those objects which are likely to be dead because of timestamp comparison, no?

Reply:
yep, done